compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -283,17 +283,28 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
           val ctx = comparerContext
           given Context = ctx // optimization for performance
           val info2 = tp2.info
+
+          /** Does `tp2` have a stable prefix?
+           *  If that's not the case, following an alias via asSeenFrom could be lossy
+           *  so we should not conclude `false` if comparing aliases fails.
+           *  See pos/i17064.scala for a test case
+           */
+          def hasStablePrefix(tp: NamedType) =
+            tp.prefix.isStable
```

Comment:
I dropped the NoPrefix check since it's already implied by isStable

Reply:
Ah yes, I overlooked that.